epsagon/trace.py

```diff
@@ -850,22 +850,38 @@ def _strip_key(key):
 
     def remove_ignored_keys(self, input_dict):
         """
-        Remove ignored keys recursively.
+        Remove ignored keys recursively in input_dict.
+        If an ignored key has been found in a dict, then the
+        dict is copied (shallow copy) and the ignored key is removed.
         :param input_dict: Input dict to remove ignored keys from.
-        :return: None
+        :return: tuple of -
+        boolean specifying if any key has been removed
+        a dict without the the ignored keys
         """
+        # pylint: disable=too-many-nested-blocks
+        copied_dict = None
         if self.keys_to_ignore:
             # Python 2 returns a list, while Python3 returns an iterator.
-            for key, value in list(input_dict.items()):
+            for key in input_dict:
                 if self._strip_key(key) in self.keys_to_ignore:
-                    input_dict.pop(key)
+                    if copied_dict is None:
+                        copied_dict = input_dict.copy()
```

Comment:
I think there's no point in actually shallow copying the data.
let's just iterate through the previous, and filter out the key we don't want